### PR TITLE
Skip validation in ClinVar submission via API

### LINKED ISSUE #3947: Remove validation buttons in ClinVar API submission form
At the same time make sure that any error returned from the submission process is clearly shown in Scout (to fix the submission and resubmit it correclty)

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR changes a functionality:
- Removes the validation step from the ClinVar submission via API - fix #3947 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. **Note that in the code I have replaced the prod submission url with the test one, so it's fine to test any submission for the purpose of this PR**
2. Locally or on test server, prepare a ClinVar submission containing one or more variants and casedata
3. Click directly on "Submit to ClinVar" even if your submission has no associated Submission ID 
4. The submission should be successful - and an official submission ID (SUBnnnnnn) should be saved in database for it
5. Alternatively the submission process should fail, but it should return the errors that occurred

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR DN

### CODE REVIEW

File: scout/constants/clinvar.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CLINVAR_API_URL = "https://submit.ncbi.nlm.nih.gov/api/v1/submissions/"
```
Comment (northwestwitch) [inline scout/constants/clinvar.py:1]: This change is for the sake of testing only, it should be reverted immediately before merging this PR!

File: tests/server/blueprints/clinvar/test_clinvar_views.py [hunk 2]
```diff
@@ -149,7 +148,7 @@ def test_delete_clinvar_object(app, institute_obj, case_obj, clinvar_form):
 
 
 def test_clinvar_update_submission(app, institute_obj, case_obj, clinvar_form):
-    """Test the endpoint resposible for updating a ClinVar submission"""
+    """Test the endpoint responsible for updating a ClinVar submission"""
```
Comment (dnil) [inline tests/server/blueprints/clinvar/test_clinvar_views.py:151]: 🏅

**Review by dnil — APPROVED** [review]
Looks good to me! 

While testing, I noticed that one can hit "submit" also on closed submissions, which does not open the actual modal, only dims the background. In contrast to a submitted submission, which can not be submitted. Upon checking why, it seems the page with the modal is in a little bit of a mess with forms dubiously without content and a couple of mismatched divs. But, this is not the topic of this PR, so feel very free to merge this when satisfied, and we can return with some html fixes later.

File: scout/server/blueprints/clinvar/controllers.py [hunk 3]
```diff
@@ -439,13 +412,27 @@ def send_api_submission(institute_id, submission_id, key):
...
+    if clinvar_id:  # Check if submission object has already an associated ClinVar ID
+        conversion_res["submissionName"] = clinvar_id
 
     code, submit_res = clinvar_api.submit_json(conversion_res, key)
 
-    if code in [200, 201, 204]:  # 204 is returned only for dry runs - used for testing
-        flash("Submission saved successfully", "success")
+    if code in [200, 201]:
+        clinvar_id = (
+            submit_res.json().get("id") if submit_res.json().get("id") != clinvar_id else clinvar_id
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/clinvar/controllers.py:426]: Hehe I was under the impression that it was an overkill! Perhaps I'll make it overwrite on the clinvar id and that's it

File: scout/server/blueprints/clinvar/templates/clinvar/clinvar_howto.html [hunk 4]
```diff
@@ -13,12 +13,7 @@ <h5>Requirements</h5>
...
-            Once you are satisfied with the content of your submission, you could validate it and obtain a submission ID from the ClinVar API by clicking on the <strong>Validate and obtain ID from ClinVar</strong> button as in the image below. This does not require an institute specific API key.<br><br>
-
-            <img width="202" alt="image" src="https://user-images.githubusercontent.com/28093618/208666249-2bfe457d-a194-4d5f-a856-97e75da14681.png"><br><br>
-
-            If the validation is successful, a submission name will be saved for your submission. Alternatively validation errors will be displayed on the page.
-            To submit your variants using the ClinVar API, you'll need to click on the <strong>Submit to ClinVar</strong> button:<br><br>
+            Once you are satisfied with the content of your submission in Scout, you can submit your variants using the ClinVar API. To use this functionality you'll need to click on the <strong>Submit to ClinVar</strong> button:<br><br>
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/clinvar/templates/clinvar/clinvar_howto.html:16]: Good idea, I'll change it like this:

image

**Comment by northwestwitch** [comment]
> While testing, I noticed that one can hit "submit" also on closed submissions, which does not open the actual modal, only dims the background. In contrast to a submitted submission, which can not be submitted. Upon checking why, it seems the page with the modal is in a little bit of a mess with forms dubiously without content and a couple of mismatched divs. But, this is not the topic of this PR, so feel very free to merge this when satisfied, and we can return with some html fixes later.

I don't want to touch it again, so I'll fix this thing as well! Nice that you noticed it!

**Comment by northwestwitch** [comment]
> > While testing, I noticed that one can hit "submit" also on closed submissions, which does not open the actual modal, only dims the background. In contrast to a submitted submission, which can not be submitted. Upon checking why, it seems the page with the modal is in a little bit of a mess with forms dubiously without content and a couple of mismatched divs. But, this is not the topic of this PR, so feel very free to merge this when satisfied, and we can return with some html fixes later.
> 
> I don't want to touch it again, so I'll fix this thing as well! Nice that you noticed it!

Fixed:

image

**Pull request merged**